### PR TITLE
Request Permissions view & HTTPError handler

### PR DESCRIPTION
**Pull request by dcloud**
**Description of change**

Users who have an account but are denied access because they have no permissions should see a message helping them request permission. On the backend, `authMiddleware` will return a '403: Forbidden' http response, so the frontend should understand this as a permissions issue (different from a '401: Unauthorized').

This PR adds an `HTTPError` custom class, which is used by the frontend to handle HTTP error codes. The `App.js` will set `authError` if an HTTP 403 error is thrown when fetching a user.

**How to test**

Set `CURRENT_USER_ID=2` in your .env file (and restart the backend): Dumbledore is a user with no permissions, so you should see the "Request Permissions" page.

Create a user with only `SITE_ACCESS`, log in as that user and you should NOT see this message. Same for users with multiple permissions (including `SITE_ACCESS`).

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/139

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Review by gopar — APPROVED** [review]
LGTM
Also had no idea you could do 
```
export {
  xxx as yyy,
  anotherThingy
}
```
Learn something new \o/

**Review by jasalisbury — APPROVED** [review]
:tada:

File: frontend/src/permissions.js [hunk 1]
```diff
@@ -64,11 +64,15 @@ export const getRegionWithReadWrite = (user) => {
...
     (p) => p.scopeId === SCOPE_IDS.READ_WRITE_ACTIVITY_REPORTS,
   ) !== undefined;
 };
 
-export default isAdmin;
+export {
+  isAdmin as default,
+  getRegionWithReadWrite,
+  hasReadWrite,
+};
```
Comment (rahearn) [inline frontend/src/permissions.js:78]: I love consolidating these here, and I didn't know that you could specify `default` as part of `export { xxx as yyy }`. Neat trick

File: frontend/src/__tests__/App.js [hunk 2]
```diff
@@ -51,8 +51,9 @@ describe('App', () => {
       render(<App />);
     });
 
-    it('displays the login button for now. in the future this should show the "request permissions" UI', async () => {
-      expect(await screen.findByText(loginText)).toBeVisible();
+    it('displays the "request permissions" page', async () => {
```
Comment (rahearn) [inline frontend/src/__tests__/App.js:54]: Hooray for finding and updating the placeholder test.

**Review by rahearn — CHANGES_REQUESTED** [review]
Everything here looks great, but can we remove the request permissions section from the unauthenticated dialog?

Screen Shot 2021-03-08 at 10 46 28 AM

I want to prevent, as much as possible, anyone filling out that form before they login the first time with HSES, since I can't assign permissions if they haven't created the User model by signing in.

**Comment by dcloud** [comment]
> Everything here looks great, but can we remove the request permissions section from the unauthenticated dialog?
> 

Removed via 9e6b293d1de344e33abb4aae341de90b73035b56

**Review by rahearn — APPROVED** [review]

**Pull request merged**